### PR TITLE
provider/aws: Add tags to Route53 hosted zones

### PR DESCRIPTION
**Pull request by catsby**
adds Tags to route53 zones

### CODE REVIEW

File: builtin/providers/aws/tags_route53_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+			t.Fatalf("%d: bad create: %#v", i, cm)
+		}
+		if !reflect.DeepEqual(rm, tc.Remove) {
+			t.Fatalf("%d: bad remove: %#v", i, rm)
+		}
+	}
+}
+
+// testAccCheckTags can be used to check the tags on a resource.
+func testAccCheckTagsR53(
```
Comment (phinze) [inline builtin/providers/aws/tags_route53_test.go]: Is this function used anywhere? I'd expect a diff on the `zone_test.go` tossing some tag stuff in there.

Comment (catsby) [inline builtin/providers/aws/tags_route53_test.go]: yup, adding acc. test now that uses it, got trigger happy :smile:

File: builtin/providers/aws/resource_aws_route53_zone.go [hunk 2]
```diff
@@ -87,9 +90,38 @@ func resourceAwsRoute53ZoneRead(d *schema.ResourceData, meta interface{}) error
 		return err
 	}
 
+	// get tags
+	req := &route53.ListTagsForResourceRequest{
+		ResourceID:   aws.String(d.Id()),
+		ResourceType: aws.String("hostedzone"),
+	}
+
+	resp, err := r53.ListTagsForResource(req)
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_route53_zone.go]: Checked the docs here - this API should not return an error when tags are empty like that weirdo S3 one. :smile:

File: builtin/providers/aws/resource_aws_route53_zone.go [hunk 3]
```diff
@@ -87,9 +90,38 @@ func resourceAwsRoute53ZoneRead(d *schema.ResourceData, meta interface{}) error
...
+	if err != nil {
+		return err
+	}
+
+	var tags []route53.Tag
+	if resp.ResourceTagSet != nil {
+		tags = resp.ResourceTagSet.Tags
+	}
+	d.Set("tags", tagsToMapR53(tags))
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_route53_zone.go]: Error check for me here? :heart_decoration:

**Comment by phinze** [comment]
LGTM - just one nitpick :cactus:

File: builtin/providers/aws/resource_aws_route53_zone.go (hunk 2 above)
Comment (catsby) [inline builtin/providers/aws/resource_aws_route53_zone.go]: Correct, it returns a `route53.ResourceTagSet` which may have a nil set of `tags`.

**Comment by phinze** [comment]
Bingo. :traffic_light: :bulb: :car: :dash:

**Pull request merged**